packages/framework/src/client/scheduler.js

```diff
@@ -122,19 +161,52 @@ export default class Scheduler {
 			return STATUS.complete !== status && STATUS.failed !== status;
 		} );
 	};
+}
 
-	/**
-	 * Checks if a request is ready.
-	 * @param {ResourceRequest} request The request to check.
-	 * @param {Date} now The current time.
-	 * @return {boolean} True if the request is ready to be sent, false otherwise.
-	 */
-	isRequestReady = ( request, now = new Date() ) => {
-		const status = request.getStatus();
-		if ( STATUS.scheduled === status || STATUS.overdue === status ) {
-			const timeLeft = request.getTimeLeft( now );
-			return ( timeLeft <= 0 );
+/**
+ * Sorts an array of requests by operation name
+ * @param {Array} requests An array of requests to be sorted
+ * @param {Date} now The current time
+ * @return {Object} The same requests as given, but in an object keyed by operation name
+ */
+export function getRequestsByOperation( requests ) {
+	return requests.reduce( ( requestsByOperation, request ) => {
```

Comment:
`return lodash.groupBy( requests, 'operation' );` ([#](https://lodash.com/docs/4.17.11#groupBy))

Maybe it's because I'm not used to it, but reading code with `Array.reduce` hurts my brain :/

Reply:
👍

Reply:
Thanks so much for this! I've never used `groupBy` before and I was able to remove this function altogether now!

Reply:
Updated in fcd3cab5d80d44d6f32d2514104adad4b82508bd